src/Hl7.FhirPath/FhirPath/Expressions/Typecasts.cs

```diff
@@ -145,6 +145,21 @@ public static bool CanCastTo(object source, Type to)
             return from == null ? to.IsNullable() : getImplicitCast(from, to) != null;
         }
 
+        public static bool IsOfType(object source, Type to)
```

Comment:
Shall we call this `IsOfExactType` - and maybe keep it `internal` for now....I don't like the duplication with `CanCastTo`, but we don't have to fix that right now.

Reply:
Happy to rename it.

Reply:
Done.